### PR TITLE
Create notifications value objects (Jira: ATQR-79)

### PR DESCRIPTION
**Pull request by raquellenn**
Implemented the value objects especified in /concept-set/Notifications.disl

Linear ATQR-127

### CODE REVIEW

**Review by rairezende — CHANGES_REQUESTED** [review]
Is your ValueObject import really working? It should've been imported by scope.

**Review by rairezende — COMMENTED** [review]
Delete empty unneeded lines and use the prettier extension to make it easier to follow it's rules (they are requested in the linear review checklist).

File: libs/atqr/domain/src/lib/notifications/notification.valueObject.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import { ValueObject } from '../../utils/valueObject';
+import { Guid } from '@tokilabs/lang';
+import { NotificationLogEntry } from './notificationLogEntry.valueObject';
+import { isBeforeDay } from '../../utils/isBeforeDay';
+
+enum NotificationCategory {
```
Comment (rairezende) [inline libs/atqr/domain/src/lib/notifications/notification.valueObject.ts]: Enums implementation from Notifications.disl are not a part of your task. It's an issue assigned to another person.

File: libs/atqr/domain/src/lib/notifications/notification.valueObject.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  constructor(
+    public readonly category: NotificationCategory,
+    public readonly templateId: MessageTemplateId,
+    public readonly allowedChannels: NotificationChannel[],
+    public readonly createdAt: Date,
+    public readonly requiresUserAction: boolean,
+    public readonly actionTaken: boolean,
+    public readonly contactLog: NotificationLogEntry[],
+    public readonly nextContactDate: Date,
+    public readonly id = new Guid()
```
Comment (rairezende) [inline libs/atqr/domain/src/lib/notifications/notification.valueObject.ts]: the "id" property should actually be the first one in the constructor, if you follow the pattern used until here.

**Review by albnunes — CHANGES_REQUESTED** [review]
it lacks a folder called valueObjects to organize it better.

remove the "valueObject" suffix from the filenames

make an index and export all the contents of that folder.

**Review by albnunes — CHANGES_REQUESTED** [review]
The isBeforeDay function is not necessary without another one that does the same

**Comment by svallory** [comment]
Replaced by #83

**Comment by svallory** [comment]
Replaced by #83

**Pull request closed without merging**